docs/ADR-51-catalyst-content-validations.md

```diff
@@ -0,0 +1,336 @@
+
+We will redefine the deployment contexts and only take into account: 'LOCAL' and 'SYNCED'.
+
+All validations will know this context and will decide if must do something different, but no validation list per context will exist, i.e. all validations will run on each deployment.
+
+Also, we will get rid of the entity version idea, moving to a timestamp based decision making on validations, so we can avoid duplicated/deprecation logic. To do so, we will be using timestamp defined in (ADR-45)[1].
+
+### Validations
+
+We will execute these validations for all entity types:
+
+- SIGNATURE
+
+  Validate that the signature belongs to the Ethereum address
+
+  ([reference](https://github.com/decentraland/decentraland-crypto/blob/master/src/Authenticator.ts))
+
+```
+function validateSignature: (see reference)
+  1. Fail if authChain is malformed
+  2. For each authLink L in authChain
+    a. Let validator V be the validator by L.type
+    b. Fail if V can't validate on that time
+
+
+1. Let validateSignature be a function that validates the auth chain of the deployment
+
+2. Fail if validateSignature(entity.id, deployment.authChain, deployment.timestamp)
+```
+
+- IPFS HASHING
+
+  Validates that all hashes used by the entity are actually IPFS hashes
+
+```
+parameter ADR_45_TIMESTAMP = 1648771200
+
+1. If deployment.timestamp <= ADR_45_TIMESTAMP return
+
+2. Let ALL_HASHES be entity.id U entity.content.hashes
+
+3. For each hash H in ALL_HASHES
+  a. Fail if H is not a valid IPFS_HASH_V1
+```
+
+- STRUCTURE
+
+  Validates that entity pointers are present and not repeated
+
+```
+1. Fail if entity.pointers.length <= 0
+
+2. Fail if Set(entity.pointers).size != entity.pointers.length
+```
+
+- METADATA SCHEMA
+  1
+  Validates entity metadata against its corresponding schema
+
+```
+parameter ADR_45_TIMESTAMP = 1648771200
+
+----
+
+1. If deployment.timestamp <= ADR_45_TIMESTAMP return
+
+2. Let entity be the uploaded deployment entity
+
+2. Let SCHEMA be the json schema of current entity.type
+
+3. Fail if entity.metadata does not satisfy SCHEMA
+```
+
+- CONTENT
+
+  Validates that uploaded and reported hashes are corrects and files corresponds to snapshots
+
+```
+parameter ADR_45_TIMESTAMP = 1648771200
+parameter ADR_X_SNAPSHOT_FILES = {"face.png", "body.png"}
+
+----
+
+1. For each content file item C in the deployment
+  a. Fail if C is not stored in disk or is not one of the uploaded files
+
+2. For each uploaded file F in the deployment
+  a. Fail if F is not one of the content file items in deployment
+
+3. If deployment.timestamp > ADR_45_TIMESTAMP
+   a. For each content file F in the deployment
+      i. Fail if F.filename ∉ ADR_X_SNAPSHOT_FILES
+```
+
+- SIZE
+
+  Validates that the full request size is within limits
+
+```
+parameter ADR_45_TIMESTAMP = 1648771200
+
+----
+
+1. Let MAX_SIZE be the maximum size in bytes allowed by ENTITY_TYPE
```

Comment:
Where is the MAX_SIZE for every ENTITY_TYPE defined?

Reply:
Here in [`catalyst-commons`](https://github.com/decentraland/catalyst-commons/blob/master/src/entities/entities.ts)

Reply:
Added current max sizes for each entity type.